eval/src/vespa/eval/eval/value_codec.h

```diff
@@ -0,0 +1,33 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "simple_value.h"
+#include <vespa/vespalib/stllike/string.h>
+
+namespace vespalib { class nbostream; }
+
+namespace vespalib::eval {
+
+/**
+ * encode a value to binary format
+ **/
+void new_encode(const Value &value, nbostream &output);
```

Comment:
I suggest we call this encode_value. Also, we could mention that this requires new api compliance in the comment.